### ISSUE TITLE
[microprofile] The /openapi does not render the additionalProperties.

### ISSUE THREAD

**Issue by charleech** [payara/Payara#3724]
# Description #
----------

I'm using the `OASFilter` to define additional schema for Map based on the [Dictionaries, HashMaps and Associative Arrays](https://swagger.io/docs/specification/data-models/dictionaries/). During accessing the `/openapi`, the `additionalProperties` is not rendered.

## Expected Outcome

```json
{
    "openapi": "3.0.0",
    "components": {
        "schemas": {
            "SimpleMap": {
                "type": "object",
                "additionalProperties": {
                    "type": "string"
                 }
            }
        }
    }
}
```

Please see the full `openapi.json` here.
<details>
<summary>The full openapi.json</summary>
<pre>
{
    "openapi": "3.0.0",
    "info": {
        "title": "Demo App",
        "description": "A Demo Application.",
        "version": "1.0"
    },
    "paths": {
      "/api/echo": {
        "post" : {
                "summary": "Echo with HTTP POST.",
                "description": "Echo with HTTP POST.",
                "operationId": "echoCreateForm",
                "requestBody": {
                    "description": "The posting data",
                    "content": {
                        "application/json": {
                            "schema": {
                                "$ref": "#/components/schemas/SimpleMap"
                            }
                        }
                    }
                },
                "responses": {
                    "200": {
                        "description": "The echo data",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/EchoResponse"
                                }
                            },
                            "application/xml": {
                                "schema": {
                                    "$ref": "#/components/schemas/EchoResponse"
                                }
                            }
                        }
                    }
                }
        }
      }
    },
    "components": {
        "schemas": {
            "EchoResponse" :{
              "type": "object",
              "properties": {
                "message" : {
                  "type": "string"
                }
              }
            },
            "SimpleMap": {
                "type": "object",
                "additionalProperties": {
                    "type": "string"
                 }
            }
        }
    }
}
</pre>
</details>

## Current Outcome

```json
{
    "openapi": "3.0.0",
    "components": {
        "schemas": {
            "SimpleMap": {
                "type": "object"
            }
        }
    }
}
```
## Steps to reproduce (Only for bug reports) 

N/A

### Samples

```java
package my.filter;

import org.eclipse.microprofile.openapi.OASFactory;
import org.eclipse.microprofile.openapi.OASFilter;
import org.eclipse.microprofile.openapi.models.OpenAPI;
import org.eclipse.microprofile.openapi.models.media.Schema;
import org.eclipse.microprofile.openapi.models.media.Schema.SchemaType;

public class DemoOASFilter implements OASFilter {

    @Override
    public void filterOpenAPI(final OpenAPI openAPI) {
        openAPI.getComponents().
                getSchemas().
                put("SimpleMap",
                    this.createMapKey());
    }
    
    private Schema createMapKey() {
        return OASFactory.createObject(Schema.class).
                          type(SchemaType.OBJECT).
                          additionalProperties(
                              OASFactory.createObject(Schema.class).
                                         type(SchemaType.STRING)
                          );         
    }
}
```

```properties
#META-INF/microprofile-config.properties
mp.openapi.filter=my.filter.DemoOASFilter
```

## Context (Optional)

I'm trying to learn the Microprofile: OpenAPI and would like to define the schema for Map.

## Environment ##

From Docker `payara/micro:latest`

- **Payara Version**: `Payara Micro  5.184 #badassmicrofish (build 89)`
- **Edition**: `Micro`
- **JDK Version**: 
```
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (IcedTea 3.9.0) (Alpine 8.181.13-r0)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```
- **Operating System**: 
```
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.8.1
PRETTY_NAME="Alpine Linux v3.8"
```
- **Database**:  N/A

**Comment by rdebusscher** [comment]
Hi Charlee,

Thank you for this report. I can confirm that the **additionalProperties** aren't 'rendered' in the output.

I have created an internal development issue with the identification `PAYARA-3509`

Thanks
Rudy

**Comment by jbee** [comment]
Added 2 tests to PR https://github.com/payara/Payara/pull/3852 that make sure this works in 1.1. Only aspect that they do not test are problems in picking up the filter configuration itself. As this is build into MP API this should work but we should verify by manually testing full end-2-end when my PR is merged.

**Comment by jbee** [comment]
Fixed by #3852. Will be released in 5.192.
